apps/trading-e2e/src/support/pages/deal-ticket-page.js

```diff
@@ -0,0 +1,115 @@
+  orderSizeField = 'order-size';
+  orderPriceField = 'order-price';
+  orderTypeDropDown = 'order-tif';
+  datePickerField = 'date-picker-field';
+  placeOrderBtn = 'place-order';
+  placeOrderErrorTxt = 'input-error-text';
+  orderDialog = 'order-wrapper';
+  orderStatusHeader = 'order-status-header';
+  orderTransactionHash = 'tx-hash';
+  orderErrorTxt = 'error-reason';
+
+  placeMarketOrder(isBuy, orderSize, orderType) {
+    cy.get(`[data-testid=${this.placeOrderBtn}]`, { timeout: 5000 }).should(
+      'be.visible'
+    );
+
+    if (isBuy == false) {
+      cy.getByTestId(this.sellOrder).click();
+    }
+
+    cy.getByTestId(this.orderSizeField).clear().type(orderSize);
+    cy.getByTestId(this.orderTypeDropDown).select(orderType);
+  }
+
+  placeLimitOrder(isBuy, orderSize, orderPrice, orderType) {
+    cy.getByTestId(this.limitOrderType).click();
+
+    if (isBuy == false) {
+      cy.getByTestId(this.sellOrder).click();
+    }
+
+    cy.getByTestId(this.orderSizeField).clear().type(orderSize);
+    cy.getByTestId(this.orderPriceField).clear().type(orderPrice);
+    cy.getByTestId(this.orderTypeDropDown).select(orderType);
+
+    if (orderType == 'GTT') {
+      const today = new Date(new Date().setSeconds(0));
+      const futureDate = new Date(today.setMonth(today.getMonth() + 1)); // set expiry to one month from now
+      const formattedDate = this.formatDate(futureDate);
+      cy.getByTestId(this.datePickerField).click().type(formattedDate);
+    }
+  }
+
+  verifyOrderRequestSent() {
+    cy.getByTestId(this.orderStatusHeader).should(
+      'have.text',
+      'Awaiting network confirmation'
+    );
+    cy.getByTestId(this.orderTransactionHash)
+      .invoke('text')
+      .should('contain', 'Tx hash: ')
+      .and('have.length.above', 64);
+  }
+
+  verifyOrderFailedInsufficientFunds() {
+    cy.getByTestId(this.orderErrorTxt).should(
+      'have.text',
+      'Reason: InsufficientAssetBalance'
+    );
+  }
+
+  clickPlaceOrder() {
+    cy.getByTestId(this.placeOrderBtn).click();
+  }
+
+  verifyPlaceOrderBtnDisabled() {
+    cy.getByTestId(this.placeOrderBtn).should('be.disabled');
+  }
+
+  verifySubmitBtnErrorText(expectedText) {
+    cy.getByTestId(this.placeOrderErrorTxt).should('have.text', expectedText);
+  }
+
+  verifyOrderRejected(errorMsg) {
+    cy.getByTestId(this.orderStatusHeader).should(
+      'have.text',
+      'Order rejected by wallet'
+    );
+    cy.getByTestId(this.orderDialog)
+      .find('pre')
+      .should('contain.text', errorMsg);
+  }
+
+  reloadPageIfPublicKeyErrorDisplayed() {
+    cy.get('body').then(($body) => {
+      if ($body.find(`[data-testid=${this.placeOrderErrorTxt}]`).length) {
+        cy.getByTestId(this.placeOrderErrorTxt)
+          .invoke('text')
+          .then(($errorText) => {
+            if ($errorText == 'No public key selected') {
+              cy.reload;
+            }
+          });
+      }
+    });
+  }
+
+  formatDate(date) {
```

Comment:
I initially did use datetime but that was erroring from the input because datetime includes _seconds_ which the date picker input doesn't like. Hence why I pulled some of the code into the page file. If there's another way to do it let me know.